ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
+        {
+            StringBuffer flags;
+            if (info.compareAllRows)
+                flags.append("|DFcompareall");
+            if (info.keepLeft)
+                flags.append("|DFkeepleft");
+            if (info.keepBest)
+                flags.append("|DFkeepbest");
+            if (flags.length()==0)
+                flags.append("|0");
+            doBuildUnsignedFunction(instance->startctx, "getFlags", flags.str()+1);
+        }
         if (!matchesConstantValue(info.numToKeep, 1))
+        {
             doBuildUnsignedFunction(instance->startctx, "numToKeep", info.numToKeep);
+            if (info.keepBest)
+                throwError(HQLERR_DedupBestWithKeepn);
```

Comment:
BEST and KEEP(n) could make sense / could be implemented I think ?
Should this have a // TODO comment ?